src/internal/connector/onedrive/item.go

```diff
@@ -67,12 +67,20 @@ func driveItemReader(
 // doesn't have its size value updated as a side effect of creation,
 // and kiota drops any SetSize update.
 func driveItemInfo(di models.DriveItemable, itemSize int64) *details.OneDriveInfo {
+	ed, ok := di.GetCreatedBy().GetUser().GetAdditionalData()["email"]
+
+	email := ""
+	if ok {
+		email = *ed.(*string)
+	}
+
```

Comment:
I believe this is incorrect. Perhaps:
```go
	user  := di.GetCreatedByUser()
	if user != nil{
             
                //  Need to verify that principal name is not empty
		ptr := user.GetUserPrincipalName()
                if ptr != nil{
                          email = *ptr
                          }
	         }
```

Reply:
Let me look into it more, but `GetCreateByUser` was always `nil` when I tested it.

Reply:
`GetCreateByUser` seems to be always `nil` but we have data in `GetCreatedBy`. I have made this change locally, but thought I would check once  again with you before pushing it upstream.
/cc @vkamra

Reply:
Let's move forward with this. Why do we need to use `GetAdditonalData` though? What does `GetUser()` return?

Reply:
`GetUser` only has `DisplayName`.